packages/sitecore-jss-react/src/components/SitecoreContext.tsx

```diff
@@ -53,7 +54,19 @@ export class SitecoreContext<ContextType = any> extends React.Component<
     };
   }
 
+  componentDidUpdate(prevProps: any) {
```

Comment:
One of the proposed and smoke tested solutions mentioned in this PBI made use of `getDerivedStateFromProps` to solve:
```
  static getDerivedStateFromProps(props: SitecoreContextProps, state: any) {
    if (props.context !== state.context) {
      return {
        context: props.context,
        setContext: state.setContext,
      };
    }
    return null;
  }
```
I can see how it might be necessary to do a `deepEqual` here as well, but overall this solution seems much simpler vs the modifications here in `SitecoreContext` and the additional `deepEqual` with `React.memo` in the sample apps.

Was this solution not working as we thought?

Reply:
@ambrauer Let me explain. We can't use `getDerivedStateFromProps` in case:
Let's imagine that we have nested component
```js
import { withSitecoreContext } from '@sitecore-jss/sitecore-jss-react';

const C = (props) => {
    const { updateSitecoreContext, context } = props;

    const onClick = () => {
        const newContext = JSON.parse(JSON.stringify(context));

        newContext.someValue = 'test';

        updateSitecoreContext(newContext);
    }

    return (
        <div onClick={}>Test</div>
    )
}

export default withSitecoreContext({ updatable: true })(C)
```

So, in case if we want to do update from nested component, we can't, because in
`getDerivedStateFromProps` we will have the same `props` and the same `state`, and our `updateSitecoreContext` will be skipped

Reply:
Ah, yes, the update throws a wrench in that approach. Thanks for the explanation!